samples/Hallo.AspNetCore.Sample/Models/Hypermedia/AddressRepresentation.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+
+namespace Hallo.AspNetCore.Sample.Models.Hypermedia
+{
+    public class AddressRepresentation : Hal<Address>,
+        IHalLinks<Address>
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public AddressRepresentation(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+        public IEnumerable<Link> LinksFor(Address resource)
+        {
+            yield return new Link(Link.Self, $"{_httpContextAccessor.HttpContext?.Request.Path}/{resource.Id}");
```

Comment:
Problem here though is that if you make a request to /people/1/addresses/1 the links in the HAL response is /people/1/addresses/1/1

Might have to accept that the you would have to have some hard wired paths eg "/people/{resource.PersonId}/addresses/{resource.Id}